packages/react-components/react-charts-preview/library/src/utilities/utilities.ts

```diff
@@ -244,42 +257,58 @@ export function createDateXAxis(
   customDateTimeFormatter?: (dateTime: Date) => string,
   useUTC?: string | boolean,
 ) {
-  const { domainNRangeValues, xAxisElement, tickPadding = 6, xAxistickSize = 6, xAxisCount = 6 } = xAxisParams;
+  const {
+    domainNRangeValues,
+    xAxisElement,
+    tickPadding = 6,
+    xAxistickSize = 6,
+    xAxisCount,
+    hideTickOverlap,
+  } = xAxisParams;
   const xAxisScale = useUTC ? d3ScaleUtc() : d3ScaleTime();
   xAxisScale
     .domain([domainNRangeValues.dStartValue, domainNRangeValues.dEndValue])
     .range([domainNRangeValues.rStartValue, domainNRangeValues.rEndValue]);
-  const xAxis = d3AxisBottom(xAxisScale).tickSize(xAxistickSize).tickPadding(tickPadding).ticks(xAxisCount);
-  if (customDateTimeFormatter) {
-    xAxis.tickFormat((domainValue: Date, _index: number) => {
+
+  let tickCount = xAxisCount ?? 6;
+  const tickFormat = (domainValue: Date, _index: number) => {
+    if (customDateTimeFormatter) {
       return customDateTimeFormatter(domainValue);
-    });
-  } else if (culture && options) {
-    xAxis.tickFormat((domainValue: Date, _index: number) => {
+    }
+    if (culture && options) {
       return domainValue.toLocaleString(culture, options);
-    });
-  } else if (timeFormatLocale) {
-    const locale: d3TimeLocaleObject = d3TimeFormatLocale(timeFormatLocale!);
-
-    xAxis.tickFormat((domainValue: Date, _index: number) => {
+    }
+    if (timeFormatLocale) {
+      const locale: d3TimeLocaleObject = d3TimeFormatLocale(timeFormatLocale!);
       return multiFormat(domainValue, locale, useUTC);
-    });
+    }
+    if (culture === undefined && tickParams.tickFormat) {
+      if (useUTC) {
+        return d3UtcFormat(tickParams.tickFormat)(domainValue);
+      } else {
+        return d3TimeFormat(tickParams.tickFormat)(domainValue);
+      }
+    }
+    return multiFormat(domainValue, undefined, useUTC);
+  };
+  if (hideTickOverlap && typeof xAxisCount === 'undefined') {
```

Comment:
ensure that the order of priority defined by the user does not get overridden

Reply:
checked it. any user defined customizations take priority over those requiring D3 utils.